auslib/db.py

```diff
@@ -341,13 +382,17 @@ def _prepareInsert(self, trans, changed_by, **columns):
         if self.onInsert:
             self.onInsert(self, "INSERT", changed_by, query, trans)
 
+        potential_signoffs = self.getPotentialSignoffs(None, columns, transaction=trans)
+        if potential_signoffs:
```

Comment:
Ah, I think see the confusion..."potential signoffs" here actually ends up being "all required signoffs that match the affected product and/or channel". These end up getting reduced to one item per role. So the "potential" part here means that not all of the items returned by getPotentialSignoffs get used, not that we may bypass signoff entirely.

As an example, if Firefox:release requires signoff, and you try to change a Firefox:release Rule without passing any signoffs in (eg: like the web layer does when it calls Rules.update), SignoffRequiredError will be raised. When a similar call is made by enactChange, any signoffs made will be passed as well, and the change will be allowed if all the required ones have been made.

Perhaps "potential signoffs" is not the best language to use here.

Does that make sense? Did I understanding your confusion?